Lib/idlelib/configdialog.py

```diff
@@ -1845,6 +1845,243 @@ def var_changed_space_num(self, *params):
 
 
+class GenPage(Frame):
+
+    def __init__(self, parent):
+        super().__init__(parent)
+        self.parent = parent
+        self.create_page_general()
+        self.load_general_cfg()
+
+    def create_page_general(self):
+        """Return frame of widgets for General tab.
+
+        Enable users to provisionally change general options. Function
+        load_general_cfg intializes tk variables and helplist using
+        idleConf.  Radiobuttons startup_shell_on and startup_editor_on
+        set var startup_edit. Radiobuttons save_ask_on and save_auto_on
+        set var autosave. Entry boxes win_width_int and win_height_int
+        set var win_width and win_height.  Setting var_name invokes the
+        default callback that adds option to changes.
+
+        Helplist: load_general_cfg loads list user_helplist with
+        name, position pairs and copies names to listbox helplist.
+        Clicking a name invokes help_source selected. Clicking
+        button_helplist_name invokes helplist_item_name, which also
+        changes user_helplist.  These functions all call
+        set_add_delete_state. All but load call update_help_changes to
+        rewrite changes['main']['HelpFiles'].
+
+        Widget Structure:  (*) widgets bound to self
+            frame
+                frame_run: LabelFrame
+                    startup_title: Label
+                    (*)startup_editor_on: Radiobutton - startup_edit
+                    (*)startup_shell_on: Radiobutton - startup_edit
+                frame_save: LabelFrame
+                    run_save_title: Label
+                    (*)save_ask_on: Radiobutton - autosave
+                    (*)save_auto_on: Radiobutton - autosave
+                frame_win_size: LabelFrame
+                    win_size_title: Label
+                    win_width_title: Label
+                    (*)win_width_int: Entry - win_width
+                    win_height_title: Label
+                    (*)win_height_int: Entry - win_height
+                frame_help: LabelFrame
+                    frame_helplist: Frame
+                        frame_helplist_buttons: Frame
+                            (*)button_helplist_edit
+                            (*)button_helplist_add
+                            (*)button_helplist_remove
+                        (*)helplist: ListBox
+                        scroll_helplist: Scrollbar
+        """
+        parent = self.parent
+        self.startup_edit = tracers.add(
+                IntVar(parent), ('main', 'General', 'editor-on-startup'))
+        self.autosave = tracers.add(
+                IntVar(parent), ('main', 'General', 'autosave'))
+        self.win_width = tracers.add(
+                StringVar(parent), ('main', 'EditorWindow', 'width'))
+        self.win_height = tracers.add(
+                StringVar(parent), ('main', 'EditorWindow', 'height'))
+
+        # Create widgets:
+        # body and section frames.
+        frame = self
```

Comment:
Delete and carefully replace '(frame,' with '(self,' in the next four statements.  (I believe that these are the only one that need it.)